src/Jackett.Common/Indexers/HDBitsApi.cs

```diff
@@ -123,14 +123,29 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
 
             var response = await MakeApiRequest("torrents", requestData);
             var releases = new List<ReleaseInfo>();
-
             foreach (JObject r in response["data"])
             {
-                var release = new ReleaseInfo();
-                release.Title = (string)r["name"];
-                release.Comments = new Uri(SiteLink + "details.php?id=" + (string)r["id"]);
-                release.Link = new Uri(SiteLink + "download.php/" + (string)r["filename"] + "?id=" + (string)r["id"] + "&passkey=" + configData.Passkey.Value);
-                release.Guid = release.Link;
+                var guid = new Uri(
+                    SiteLink + "download.php/" + (string)r["filename"] + "?id=" + (string)r["id"] + "&passkey=" +
+                    configData.Passkey.Value);
+                var seeders = (int)r["seeders"];
+
+                var release = new ReleaseInfo
+                {
+                    Title = (string)r["name"],
+                    Comments = new Uri(SiteLink + "details.php?id=" + (string)r["id"]),
+                    Link = guid,
+                    Category = MapTrackerCatToNewznab((string)r["type_category"]),
+                    Size = (long)r["size"],
+                    Files = (long)r["numfiles"],
+                    Grabs = (long)r["times_completed"],
+                    Seeders = seeders,
+                    PublishDate = DateTimeUtil.UnixTimestampToDateTime((int)r["utadded"]),
```

Comment:
As I said in other comment, make the conversions that can fail outside the initializer.

Reply:
Does that include all the `(string)` and `(long)` in this one?

Reply:
Keep casting inside the initializer is ok for me.